src/main/java/mods/eln/sixnode/lampsocket/LightBlockEntity.java

```diff
@@ -135,29 +135,30 @@ public void updateEntity() {
             }
         }
 
-        if (light != worldObj.getBlockMetadata(xCoord, yCoord, zCoord)) {
-
-            worldObj.setBlockMetadataWithNotify(xCoord, yCoord, zCoord, light, 2);
-            worldObj.updateLightByType(EnumSkyBlock.Block, xCoord, yCoord, zCoord);
+        if (light != worldObj.getBlockState(this.pos).getBlock().getMetaFromState(worldObj.getBlockState(this.pos))) {
```

Comment:
Hold on a sec. At a minimum, you're fetching the block state twice. Try not to do that.

Reply:
You're not explaining *what* is TODO. Remember, the you in the future (or me) won't know what you're thinking.